sympy/functions/elementary/piecewise.py

```diff
@@ -223,6 +224,9 @@ def _eval_interval(self, sym, a, b):
             values = []
             intervals = self._sort_expr_cond(
                 sym, S.NegativeInfinity, S.Infinity)
+            if len(intervals) == 1 and intervals[0][-1] == self:
```

Comment:
Note to self: Some of the failing Piecewise integration tests (from issues) involve intervals in which the expressions are Piecewise, so perhaps this is just a special case of that and needs to be handled by an evaluation of that piecewise like is handled below for this special case.